active/gridsyntax/GridSyntaxSpec2.md

```diff
@@ -0,0 +1,368 @@
+    <Grid.RowDefinitions>
+          <RowDefinition Height="1*" />
+          <RowDefinition Height="Auto" />
+          <RowDefinition Height="25" />
+          <RowDefinition Height= "14" />
+          <RowDefinition Height="20" />
+    </Grid.RowDefinitions>
+</Grid>
+```
+
+The code in that example creates the ColumnDefinition objects and RowDefinition objects and adds them to the Grid's [Grid.ColumnDefinitions](https://docs.microsoft.com/uwp/api/Windows.UI.Xaml.Controls.Grid.ColumnDefinitions) and [Grid.RowDefinitions](https://docs.microsoft.com/uwp/api/Windows.UI.Xaml.Controls.Grid.RowDefinitions) collections, respectively.  This is currently the only way to set these properties in markup.
+
+The new features in this spec allow for this equivalent syntax:
+
+```xml
+<Grid ColumnDefinitions="1*, 2*, Auto, *, 300" RowDefinitions="1*, Auto, 25, 14, 20" />
+```
+
+This new syntax is enabled by a new Xaml language feature and associated APIs that can be used on any collection-typed property, and includes updates to Grid to use these APIs.
+
+# Description
+
+We've introduced a new XAML language feature that allows the initialization of collection-type properties (including read-only properties) using Xaml attribute syntax. We've also updated the Grid APIs to make sure that this syntax works properly in particular for Grid's use cases.
+
+Grid initialization in Xaml markup has always required a lot of XML:
+
+```xml
+<Grid>
+    <Grid.ColumnDefinitions>
+          <ColumnDefinition Width="Auto" />
+          <ColumnDefinition Width="*" />
+    </Grid.ColumnDefinitions>
+</Grid>
+```
+
+You can now accomplish the same thing in a single tag:
+
+```xml
+<Grid ColumnDefinitions="Auto, *">
+```
+
+Before this change, if you write that shorter Xaml markup, the  Xaml loader would try to convert the string `Auto, *` into a [ColumnDefinitionCollection](https://docs.microsoft.com/uwp/api/Windows.UI.Xaml.Controls.ColumnDefinitionCollection) object, as that's the type of the [Grid.ColumnDefinitions](https://docs.microsoft.com/uwp/api/Windows.UI.Xaml.Controls.Grid.ColumnDefinitions) property. That fails because there is no such conversion available. (And it wouldn't help if there was, because the property is read-only.)
+
+But the Xaml syntax rules now support this scenario by recognizing that the target property is a collection type, and that there's no conversion from string available, and then by treating the attribute value as a comma-separated list of collection items. The items of the collection are of type [ColumnDefinition](https://docs.microsoft.com/uwp/api/Windows.UI.Xaml.Controls.ColumnDefinition), so a check is made to see if strings can be converted to ColumnDefinition instances. Since they can, the Xaml loader creates those instances and adds them to the collection.
+
+| Do we have documentation that explains what exactly a "collection" is? Can't remember if it's just IVector, or if it has more rules like the C# initialization syntax.
```

Comment:
IVector is the WinRT type and IList is the .NET projection: 
- https://docs.microsoft.com/en-us/uwp/api/windows.foundation.collections.ivector-1?view=winrt-19041 
https://docs.microsoft.com/en-us/windows/uwp/cpp-and-winrt-apis/collections

Reply:
This link also gives an overview of collections in .NET: https://docs.microsoft.com/en-us/uwp/api/windows.foundation.collections?view=winrt-19041

Reply:
C# has a definition of "collection" that's not defined by an interface (in the initialization syntax). For example, among other things, it looks for an "Add" method with a particular signature. IList has a matching Add method, so you can implement that and it works. (Similarly C# 'foreach' looks for a method name/signature, not IEnumerable.) 

I think the answer is that you're right, the WinUI Xaml compiler/loader doesn't have a definition like that, and relies specifically on IVector/IList.